src/MIDebugEngine/AD7.Impl/AD7Engine.cs

```diff
@@ -824,10 +831,13 @@ public int ExecuteOnThread(IDebugThread2 pThread)
         // that is, not all threads should be required to be stopped before this method returns. The implementation of this method may be 
         // as simple as calling the IDebugProgram2::CauseBreak method on this program.
         //
-        // The sample engine only supports debugging native applications and therefore only has one program per-process
         public int Stop()
         {
-            throw new NotImplementedException();
+            DebuggedProcess.WorkerThread.RunOperation(async () =>
+            {
+                await _debuggedProcess.CmdBreak();
```

Comment:
I don't believe we depend on the timing in any way for the child processing case. As far as I can tell the debugger makes no assumption about process state based on this call. It would only be if the UI were to make some assumptions. But the ui seems to wait for notification from the separate processes correctly.